data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexManagerFactory.java

```diff
@@ -8,32 +8,59 @@
 import org.opensearch.client.RestHighLevelClient;
 import org.opensearch.dataprepper.plugins.sink.opensearch.OpenSearchSinkConfiguration;
 
+import java.io.IOException;
 import java.util.Optional;
 
 public class IndexManagerFactory {
 
-    public final IndexManager getIndexManager(final IndexType indexType,
+    public final AbstractIndexManager getIndexManager(final IndexType indexType,
                                         final RestHighLevelClient restHighLevelClient,
                                         final OpenSearchSinkConfiguration openSearchSinkConfiguration) {
+	try {
+	    return (AbstractIndexManager) getIndexManager(indexType, restHighLevelClient, openSearchSinkConfiguration, null);
+	} catch (IOException e) {
+	    return null;
+	}
+    }
+
+    public final IndexManager getIndexManager(final IndexType indexType,
+                                        final RestHighLevelClient restHighLevelClient,
+                                        final OpenSearchSinkConfiguration openSearchSinkConfiguration,
+					String indexAlias) throws IOException {
+        if (indexAlias != null && isDynamicIndexAlias(indexAlias)) {
+	    return  new DynamicIndexManager(indexType, restHighLevelClient, openSearchSinkConfiguration);
```

Comment:
You can add a parameter to the `DynamicIndexManager` constructor which takes an `IndexManagerFactory`. Then you can pass `this` in here.

```
return new DynamicIndexManager(indexType, restHighLevelClient, openSearchSinkConfiguration, this);
```